### ISSUE TITLE
False positive `explicit_iter_loop` for &mut [T; N]

### ISSUE THREAD

**Issue by Zarathustra30** [Manishearth/rust-clippy#261]
```
fn main() {
    let mut bodies_array = [1,2,3,4,5];
    offset_momentum(&mut bodies_array)
}

fn offset_momentum(bodies: &mut [i32;5]) {
    for bi in bodies.iter() {
        println!("{}", bi);
    }
    bodies[2] = 7;
}
```
Produces the `explicit_iter_loop` warning, but replacing `bodies.iter()` with `&bodies` returns 
>error: the trait `core::iter::Iterator` is not implemented for the type `&&mut [i32; 5]`

and suggests "calling `.iter()` or a similar method"

Note: in this case, `iter()` is required, otherwise the for-loop consumes `bodies` and renders it unusable.

**Comment by llogiq** [comment]
There is an 
`impl<'a, T> IntoIterator for &'a mut [T; 5]` in std, so it should be possible to simply remove the `.iter()` and use bodies directly (this will however take `self` (the mut pointer) by value (thus subsequent usage won't work if I'm not mistaken).

**Comment by Zarathustra30** [comment]
Ah. Yes. There is a borrow after the for loop in the original ([n-body benchmark](http://benchmarksgame.alioth.debian.org/u64q/program.php?test=nbody&lang=rust)). Stupid minimal examples. I think this is still technically a false positive, though, so I fixed the example.

**Comment by llogiq** [comment]
Even if it wasn't a false positive, the suggestion made by the lint was wrong. Thanks for spotting this and taking the time to find a reduced example.
